### PR TITLE
Tagging feature for "belongs to many" relations

### PR DESCRIPTION
**Pull request by amrnn90**
This PR adds an option for belongsToMany relationships that enables the user to create new items directly using select2 and ajax.
Before using this, it's necessary to enable BREAD for the other table and make fields other than "label" nullable or give them default values.

### CODE REVIEW

**Review by marktopper — APPROVED** [review]

**Comment by marktopper** [comment]
Thanks, works like a charm
